### PR TITLE
AV-196885 documentation for L7CRD

### PR DESCRIPTION
**Pull request by arihantg**

### CODE REVIEW

File: docs/crds/hostrule.md [hunk 1]
```diff
@@ -278,6 +279,9 @@ The `loadBalancerIP` field can be used to provide a valid preferred IPv4 address
 
 **Note**: The HostRule CRD is not aware of the misconfigurations while it is being created, therefore the HostRule will be `Accepted` nonetheless.
 
+#### L7Rule 
+
+L7rule field can be used to specify the name of [L7Rule](./l7rule.md) Crd whose properties needs to be applied on the VS created through Hostrule.
```
[the remark this answers is not in the document]
Comment (akshayhavile) [reply docs/crds/hostrule.md:284]: `Crd whose properties needs to be applied on the VS created through Hostrule.` Can we rephrase it? This sounds like hostrule creations results in VS creation

File: docs/crds/l7rule.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+### L7Rule 
+
+L7Rule CRD can be used to modify the default properties of the L7 VS created using the Hostrule CRD. [`AKOSettings.enableEVH`](../akoconfig.md#akoconfig-custom-resource) needs to be set to `true` to enable this feature.
```
Comment (akshayhavile) [inline docs/crds/l7rule.md:3]: There is empty union between Hostrule and L7 rule properties.  We can say that it is used to override or modify VS properties which are not part of Hostrule.

Comment (akshayhavile) [reply docs/crds/l7rule.md:3]: L7Rule is applicable when AKO is running in EVH mode.

File: docs/crds/l7rule.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+```yaml
+  status:
+    error: botPolicyRef "My-L7-Application" not found on controller
+    status: Rejected
+```
+
+#### Conditions and Caveats
+
+##### L7Rule deletion
```
Comment (akshayhavile) [inline docs/crds/l7rule.md:104]: Should L7 Rule and hostrule be in the same namespace?

Comment (akshayhavile) [reply docs/crds/l7rule.md:104]: What happens if hostrule deleted and not L7 Rule?

File: docs/crds/l7rule.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  hostNameXlate: host.com
+  minPoolsUp: 2
+  performanceLimits:
+          maxConcurrentConnections: 2000
+          maxThroughput: 3000
+  securityPolicyRef: secPolicy
+  trafficCloneProfileRef: tcp
+```
+
+**NOTE**: The L7Rule CRD must be configured in the same namespace as hostrule.
```
Comment (pkoshtavmware) [inline docs/crds/l7rule.md:30]: hostrule -> HostRule

File: docs/crds/l7rule.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Specific usage of L7Rule CRD
+
+L7Rule CRD can be created to set some of the defaultproperties in a L7 virtual service.
```
Comment (pkoshtavmware) [inline docs/crds/l7rule.md:34]: defaultproperties -> default properties

Comment (pkoshtavmware) [reply docs/crds/l7rule.md:34]: virtual service -> VirtualService

File: docs/crds/l7rule.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+| `removeListeningPortOnVsDown`            | Remove listening port if VirtualService is down.                                                                                         | False |
+| `sslSessCacheAvgSize`            | Expected number of SSL session cache entries (may be exceeded).Allowed values are 1024-16383.                                                                                         | 1024 |
+| `botPolicyRef`            | Bot detection policy for the Virtual Service. It is a reference to an object of type BotDetectionPolicy                                                                                    | Nil |
+| `hostNameXlate`                         | Translate the host name sent to the servers to this value.Translate the host name sent from servers back to the value used by the client. It is not applied on child vs                                                                                                   | Nil                                   |
```
Comment (pkoshtavmware) [inline docs/crds/l7rule.md:46]: host name -> HostName

File: docs/crds/l7rule.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+| `securityPolicyRef`         | Security policy applied on the traffic of the Virtual Service. This policy is used to perform security actions such as Distributed Denial of Service (DDoS) attack mitigation, etc. It is a reference to an object of type SecurityPolicy and is not applied on child vs.                                                                                       |   Nil                            |
+| `trafficCloneProfileRef`          | Server network or list of servers for cloning traffic. It is a reference to an object of type TrafficCloneProfile.                                                                                     | Nil |
+
+
+
+#### Attaching L7Rule to HostRule
+
+An L7Rule can be specifed in hostrule spec. Respective L7Rule Properties will be applied to the VS created through corresponding Hostrule. An L7Rule can be attached in the Hostrule CRD spec with `l7Rule` as the key and `name of the l7rule crd` as the value.
```
Comment (pkoshtavmware) [inline docs/crds/l7rule.md:57]: hostrule -> HostRule
(Proper Noun)

File: docs/crds/l7rule.md (hunk 2 above)
Comment (arihantg) [reply docs/crds/l7rule.md]: changed

File: docs/crds/l7rule.md (hunk 3 above)
Comment (arihantg) [reply docs/crds/l7rule.md:104]: they have to be in same namespace . 
On hostrule deletion L7Rule Properties will be also reset to default in VS

File: docs/crds/l7rule.md (hunk 4 above)
Comment (arihantg) [reply docs/crds/l7rule.md]: changed

File: docs/crds/l7rule.md (hunk 6 above)
Comment (arihantg) [reply docs/crds/l7rule.md]: changed

File: docs/crds/hostrule.md [hunk 8]
```diff
@@ -278,6 +279,9 @@ The `loadBalancerIP` field can be used to provide a valid preferred IPv4 address
 
 **Note**: The HostRule CRD is not aware of the misconfigurations while it is being created, therefore the HostRule will be `Accepted` nonetheless.
 
+#### L7Rule 
+
+L7rule field can be used to specify the name of [L7Rule](./l7rule.md) CRD. It is used to modify select VS Properties which are not part of HostRule CRD.
```
Comment (akshayhavile) [inline docs/crds/hostrule.md:284]: Lets add comment saying it is applicable only in `v1beta1` hostrule definition

File: docs/crds/l7rule.md (hunk 3 above)
Comment (akshayhavile) [reply docs/crds/l7rule.md:104]: @arihantg Lets mention: Hostrule and L7Rule should be in the same namespace.

File: docs/crds/hostrule.md (hunk 8 above)
Comment (arihantg) [reply docs/crds/hostrule.md:284]: added

**Review by pkoshtavmware — DISMISSED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**